### PR TITLE
Fix interpolate tracing

### PR DESCRIPTION
**Pull request by zdevito**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#19034 Fix interpolate tracing**

Summary: when the scale factor is convertible to an integer, it fails
with floor_vml_cpu not implemented for 'Long' because the expression is
a long tensor where the floor operator is a no-op.

Test Plan: test_jit.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D14837282](https://our.internmc.facebook.com/intern/diff/D14837282)

### CODE REVIEW

**Review by eellison — APPROVED** [review]
Thanks for fix!

**Review by jamesr66a — APPROVED** [review]
makes sense

File: torch/nn/functional.py [hunk 1]
```diff
@@ -2495,7 +2495,7 @@ def _output_size(dim):
 
         # make scale_factor a tensor in tracing so constant doesn't get baked in
         if torch._C._get_tracing_state():
-            return [(torch.floor(input.size(i + 2) * torch.tensor(scale_factors[i]))) for i in range(dim)]
+            return [(torch.floor(input.size(i + 2) * torch.tensor(float(scale_factors[i])))) for i in range(dim)]
```
Comment (apaszke) [inline torch/nn/functional.py]: Why is this necessary again? It looks quite benign, so if it ran in eager mode correctly, then it should run under the JIT as well I think?

Comment (apaszke) [reply torch/nn/functional.py]: How do we get to the call of `floor_vml_cpu`?
